daemon/CryptoImpl/cryptopluginfunctionwrappers.cpp

```diff
@@ -0,0 +1,305 @@
+}
+
+KeyResult CryptoPluginWrapper::generateKey(
+        CryptoPlugin *plugin,
+        const Key &keyTemplate,
+        const KeyPairGenerationParameters &kpgParams,
+        const KeyDerivationParameters &skdfParams)
+{
+    Key key;
+    Result result = plugin->generateKey(
+                keyTemplate, kpgParams, skdfParams, &key);
+    return KeyResult(result, key);
+}
+
+KeyResult CryptoPluginWrapper::generateAndStoreKey(
+        CryptoPlugin *plugin,
+        const Key &keyTemplate,
+        const KeyPairGenerationParameters &kpgParams,
+        const KeyDerivationParameters &skdfParams)
+{
+    Key keyReference;
+    Result result = plugin->generateAndStoreKey(
+                keyTemplate, kpgParams, skdfParams, &keyReference);
+    return KeyResult(result, keyReference);
+}
+
+KeyResult CryptoPluginWrapper::storedKey(
+        CryptoPlugin *plugin,
+        const Key::Identifier &identifier,
+        Key::Components keyComponents)
+{
+    Key key;
+    Result result = plugin->storedKey(
+                identifier, keyComponents, &key);
+    return KeyResult(result, key);
+}
+
+IdentifiersResult CryptoPluginWrapper::storedKeyIdentifiers(
+        CryptoPlugin *plugin)
+{
+    QVector<Key::Identifier> identifiers;
+    Result result = plugin->storedKeyIdentifiers(&identifiers);
+    return IdentifiersResult(result, identifiers);
+}
+
+DataResult CryptoPluginWrapper::calculateDigest(
+        CryptoPlugin *plugin,
+        const QByteArray &data,
+        std::tuple<Sailfish::Crypto::CryptoManager::SignaturePadding,
+                   Sailfish::Crypto::CryptoManager::DigestFunction> options)
+{
+    QByteArray digest;
+    Result result = plugin->calculateDigest(
+                data, std::get<0>(options), std::get<1>(options), &digest);
+    return DataResult(result, digest);
+}
+
+DataResult CryptoPluginWrapper::sign(
+        CryptoPlugin *plugin,
+        const QByteArray &data,
+        const Key &key,
+        std::tuple<Sailfish::Crypto::CryptoManager::SignaturePadding,
+                   Sailfish::Crypto::CryptoManager::DigestFunction> options)
+{
+    QByteArray signature;
+    Result result = plugin->sign(
+                data, key, std::get<0>(options), std::get<1>(options), &signature);
+    return DataResult(result, signature);
+}
+
+ValidatedResult CryptoPluginWrapper::verify(
+        CryptoPlugin *plugin,
+        const QByteArray &signature,
+        const QByteArray &data,
+        const Key &key,
+        std::tuple<Sailfish::Crypto::CryptoManager::SignaturePadding,
+                   Sailfish::Crypto::CryptoManager::DigestFunction> options)
+{
+    bool verified = false;
+    Result result = plugin->verify(
+                signature, data, key, std::get<0>(options), std::get<1>(options), &verified);
+    return ValidatedResult(result, verified);
+}
+
+TagDataResult CryptoPluginWrapper::encrypt(
+        CryptoPlugin *plugin,
+        std::tuple<QByteArray, QByteArray> dataAndIv,
```

Comment:
`tuple` of two arguments is never better than the `std::pair`. And a struct with named fields is always more readable over `std::pair`.